Gems/Atom/Tools/MaterialEditor/Code/Source/Viewport/MaterialViewportWidget.cpp

```diff
@@ -69,9 +69,15 @@ namespace MaterialEditor
 
         // The viewport context created by AtomToolsFramework::RenderViewportWidget has no name.
         // Systems like frame capturing and post FX expect there to be a context with DefaultViewportContextName
-        auto viewportContextManager = AZ::Interface<AZ::RPI::ViewportContextRequestsInterface>::Get();
-        const AZ::Name defaultContextName = viewportContextManager->GetDefaultViewportContextName();
-        viewportContextManager->RenameViewportContext(GetViewportContext(), defaultContextName);
+        // ------------------------------
```

Comment:
[Comment about a code below this, but since it only allows me to comment on modifications I put the comment here]

````
AZStd::shared_ptr<AzFramework::Scene> mainScene = sceneSystem->GetScene(AzFramework::Scene::MainSceneName);
````
This class uses the main scene, which is fine to use by the standalone application, but when running this within the editor I think the main scene is already in used. So when O3DE Material Editor closes it tries later to stop unset the scene and it asserts that it's not the expected scene.

@gadams3 any comment on how to approach this?
Maybe it should use another scene which is not the main one?

Reply:
It uses the main viewport and scene because there was only one in the material editor. It could be updated to use a unique scene and unique name as long as all of the systems that need a default scene continue to work. That would have to change anyway if there was a need to support multiple viewports.

Reply:
Thanks, I'll take a look and try to make it use a unique scene and unique name.